nixos/config: move nano to defaultPackages

samuelgrf opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
Some people already have another editor installed and may want to get rid of applications they don't use.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

samuelgrf committed:
nixos/config: move nano to defaultPackages

Some people already have another editor installed and may want to
get rid of applications they don't use.

samuelgrf commented:
I just noticed typos in the option description, should I just amend these to this pr or open a new one?
```diff
diff --git a/nixos/modules/config/system-path.nix b/nixos/modules/config/system-path.nix
index e317a3a1ced..b366db2dfc9 100644
--- a/nixos/modules/config/system-path.nix
+++ b/nixos/modules/config/system-path.nix
@@ -74,7 +74,7 @@ in
         default = defaultPackages;
         example = literalExample "[]";
         description = ''
-          Set of packages users expect from a minimal linux istall.
+          Set of packages users expect from a minimal Linux install.
           Like systemPackages, they appear in
           /run/current-system/sw.  These packages are
           automatically available to all users, and are
```

davidak commented:
Thanks!

>I just noticed typos in the option description, should I just amend these to this pr or open a new one?

you can use this

davidak commented on nixos/doc/manual/release-notes/rl-2009.xml:
```diff
@@ -1050,7 +1050,7 @@ services.transmission.settings.rpc-bind-address = "0.0.0.0";
    <listitem>
     <para>
-      The option <option>defaultPackages</option> was added. It installs the packages <package>perl</package>, <package>rsync</package> and <package>strace</package> for now. They were added unconditionally to <option>systemPackages</option> before, but are not strictly necessary for a minimal NixOS install. You can set it to an empty list to have a more minimal system. Be aware that some functio ...
+      The option <option>defaultPackages</option> was added. It installs the packages <package>nano</package>, <package>perl</package>, <package>rsync</package> and <package>strace</package> for now. They were added unconditionally to <option>systemPackages</option> before, but are not strictly necessary for a minimal NixOS install. You can set it to an empty list to have a more minimal system. B ...
```
this has to go in the changelog of 21.03 (which might not exist yet)

samuelgrf replied:
Can't we just backport this pr? Nano is still installed by default, so this shouldn't affect any setups.

davidak replied:
Yes, we could.

I want that defaultPackages can really be set to empty without causing any issues, so i would like more testing on unstable...

In this case it might be ok.

I have micro installed and use it most of the time, but git still opens nano. I haven't made the effort to change that yet. Have you tested for example such a case? And there might be edgecases like with rsync and nixops.

aanderse replied:
I agree with @davidak - I'm a big fan of removing packages causing side effects but let's keep this in `unstable` to bake for the next 6 months... we'll catch most issues that way.

samuelgrf replied:
Thanks for the thorough answers. I didn't think about the edge cases, so stabilizing this first makes sense, I'll update the pr.
Also, can someone proofread the release notes? English isn't my first language.

samuelgrf commented:
In the last force-push I've rewritten the option description and added a note about installing another editor and setting the corresponding environment variable.
Git needs this for writing commit messages, else it throws an error.

davidak reviewed (approved):
not tested, but looks good

samuelgrf commented:
@davidak I've fixed the merge conflict and improved some of the wording, can you take another look and merge if it looks good?
I want this to get some testing on unstable, as we are getting close to 21.05's release.

davidak reviewed (approved):
Looks good!

This should not cause any issue. Maybe when someone set's the option to empty list, it could cause unexpected behavior.

If this causes bigger issues, it can get reverted before the release.

davidak commented:
Result of `nixpkgs-review pr 97565` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
    <li>tests.nixos-functions.nixosTest-test</li>
  </ul>
</details>

davidak commented:
Result of `nixpkgs-review pr 97565` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 test built:</summary>
  <ul>
    <li>nixosTests.env</li>
  </ul>
</details>
